Makefile

```diff
@@ -1,152 +1,245 @@
+
+docker-running:
+	@docker inspect -f '{{.State.Running}}' ${PROJECT}-{db,php,web} &>/dev/null
+
+wait-healthy:
+	@echo "Wait for all containers to become healthy"
+	@python $(CURDIR)/bin/docker-compose-wait.py
+
+docker-rebuild: docker-stop # Update docker images if there has been changes to Dockerfiles
 	docker-compose -f ${DOCKER_COMPOSE_FILE} up -d --build
 	docker-compose -f ${DOCKER_COMPOSE_FILE} ps
-	@sleep 10
 
-docker-status:
+status: docker-status # Alias to docker-status
+docker-status: # Display status of containers related to this project
 	docker-compose -f ${DOCKER_COMPOSE_FILE} ps
 
-docker-start:
+start: docker-start # Alias to docker-start
+docker-start: # Start containers for this project
 	docker-compose -f ${DOCKER_COMPOSE_FILE} up -d
 	docker-compose -f ${DOCKER_COMPOSE_FILE} ps
-	@sleep 10
 
-docker-stop:
+stop: docker-stop # Alias to docker-stop
+docker-stop: # Stop containers for this project
 	docker-compose -f ${DOCKER_COMPOSE_FILE} down
 
-docker-restart: docker-stop docker-start
+docker-restart: docker-stop docker-start # Restart containers for this project
 
-composer-install:
-	composer install --ignore-platform-reqs --no-interaction --no-progress
+drush: # Forwards to drush inside php container
+	docker exec -i ${PROJECT}-php drush $(filter-out $@,$(MAKECMDGOALS))
 
-composer-update:
-	composer update --ignore-platform-reqs --no-interaction --no-progress --prefer-dist
+composer: # Runs composer
+	docker run \
+	  --rm \
+	  -w "/var/www" \
+	  -v $(CURDIR)/composer.json:/app/composer.json \
+	  -v $(CURDIR)/composer.lock:/app/composer.lock \
+	  composer $(filter-out $@,$(MAKECMDGOALS))
+	docker-compose -f ${DOCKER_COMPOSE_FILE} up -d --build
+
+composer-install: vendor # Installs composer packages from composer.lock file
+	docker exec \
+	  -i \
+	  -w "/var/www" \
+	  ${PROJECT}-php \
+		composer install \
+		  --ignore-platform-reqs \
+		  --no-interaction \
+		  --no-progress
+
+
+composer-update: # Update lock file from composer.json and rebuild images
+	docker exec \
+	  -i \
+	  --rm \
+	  -w "/var/www" \
+	  ${PROJECT}-php \
+		composer update \
+		  --lock \
+		  --no-scripts \
+		  --no-autoloader
+	docker-compose -f ${DOCKER_COMPOSE_FILE} up -d --build
 
-drupal-upgrade:
-	composer update drupal/core --with-dependencies
+drupal-upgrade: # update drupal core
+	docker exec \
+	  -i \
+	  -w "/var/www" \
+	  ${PROJECT}-php \
+		composer update drupal/core \
+		  --lock \
+		  --with-all-dependencies
+	docker-compose -f ${DOCKER_COMPOSE_FILE} up -d --build
 
-drupal-install:
-	./bin/drush --root=/var/www/web site-install minimal -vv --yes \
-		--account-name=admin \
-		--account-pass=admin \
-		--site-name="Drupal Dev Docker" \
-		install_configure_form.enable_update_status_module=NULL \
-		install_configure_form.enable_update_status_emails=NULL
+vendor:
+	mkdir $(CURDIR)/vendor
 
-config-init:
+destroy: # take town and remove all data related to this project's current state
```

Comment:
destroy: docker-destroy composer-purge pl-purge
	sudo rm -rf ./web/sites/default/files/*

Reply:
Make drupal files not a volume